### PR TITLE
Distiller fix

### LINKED ISSUE #514: Pretty print precedence for interval expressions
image

Should have parenthesis

### LINKED ISSUE #515: Pretty print for pathapp/pathlam is huacked up
`PathApp(Lam(x, a), b)` rendered as `\x. a b` what the

### PR DESCRIPTION
**Pull request by wsx-ucb**
Attempt to fix #514 and fix #515.

### CODE REVIEW

**Comment by wsx-ucb** [comment]
Fixing the issue requires passing around the appropriate `Outer` parameter.
I find it more convenient to achieve it using non-static method and calling `term` directly.
Are there any reason why helper methods like `formula`, `partial`, `cube` are declared as static previously?

**Comment by ice1000** [comment]
@wsx-ucb I think it's fine to instance method them. Have you tried f (a and b and c)? Will there be a paren

**Comment by ice1000** [comment]
@wsx-ucb > I'll then try to do the same thing for partial, cube, etc.? 

I think it unnecessary.

And yes I think we should. You may want to add some enum variants to Prec and don't worry about that. It's very flexible interface

**Comment by ice1000** [comment]
@ice1000 Sorry it should be Outer not Prec

**Comment by imkiva** [comment]
> Are there any reason why helper methods like `formula`, `partial`, `cube` are declared as static previously?

No specific reasons, but I guess that's just because we want them to compile at that time

**Comment by wsx-ucb** [comment]
> You may want to add some enum variants to Outer and don't worry about that.

Is it really this simple?
Currently `Outer` is used in `checkParen` such that only when we have strictly looser bindings the parenthesis would be omitted.
Also, the `Outer` is setup such that only total ordering between binding mode is allowed.
So we must choose whether `/\` or `\/` binds stronger.

**Comment by ice1000** [comment]
@wsx-ucb Yes, and binds stronger. I think we can assume that.

**Comment by wsx-ucb** [comment]
The desired behavior for `/\` and `\/` should be implemented now.
There are still some todos:
- [ ] Look into #515.
- [ ] Add tests for the new behavior. (Not sure how to do this.)

**Comment by ice1000** [comment]
@wsx-ucb For test you can just use assertions like Distiller Test

File: base/src/main/java/org/aya/distill/BaseDistiller.java [hunk 1]
```diff
@@ -124,6 +124,7 @@ private <T extends AyaDocile> Doc visitArg(@NotNull Fmt<T> fmt, @NotNull Arg<T>
   }
 
   public static @NotNull Doc checkParen(@NotNull Outer outer, @NotNull Doc binApp, @NotNull Outer binOp) {
+    if (outer == binOp && (outer == Outer.IMin || outer == Outer.IMax)) return binApp;
```
Comment (ice1000) [inline base/src/main/java/org/aya/distill/BaseDistiller.java:127]: There's a way to avoid this

Comment (wsx-ucb) [reply base/src/main/java/org/aya/distill/BaseDistiller.java:127]: Oh please let me know!
I admit my solution is a bit hacky.

**Pull request merged**